src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -91,7 +91,8 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
                 }
                 else if (SystemEnvironment.Instance.IsLinuxAppService())
                 {
-                    return new LinuxAppServiceEventGenerator(new LinuxAppServiceFileLoggerFactory());
+                    var hostNameProvider = p.GetService<HostNameProvider>();
```

Comment:
This could be cleaned up... but let's not do that as part of this changes. We can follow up on changes to how we load the  appropriate event generator